### ISSUE TITLE
Help tooltip for the response/output mapping in connectors

### ISSUE THREAD

**Issue by christian-konrad** [camunda/connectors#154]
__Is your feature request related to a problem? Please describe.__

The response mapping in Connector templates/tasks is not self-explaining. A new user might drop off here, since it's completely different to how it works in other tasks, and there is no in-app guide. The UI is lacking following crucial information:
* Name of the variables (`status`, `header`, `body`) that they can map
* How to write the FEEL context to "destructure"/map the response

Besides that, the mapped variables are not available in variable autocompletion in the process after declaration with this FEEL context.

__Describe the solution you'd like__

Provide autocompletion for the variables, as well as helpful tooltips. 

![connector-properties-2](https://user-images.githubusercontent.com/56470362/209305324-85fb8ed7-a9a9-4236-8a58-b95a9c391ba3.png)

Related to https://github.com/bpmn-io/properties-panel/issues/202

__Describe alternatives you've considered__

Might even be a better (but major breaking) solution:

Use output variable mapping as people are already used to, and provide the response variables (`status`, `header`, `body`) in the autocompletion there.

![connector-properties](https://user-images.githubusercontent.com/56470362/209300993-bf298c52-d03f-44e3-8a94-60fc7e90d357.png)

__Additional context__

**Comment by christian-konrad** [comment]
@bastiankoerber @nikku fyi

**Comment by nikku** [comment]
Regarding this

> How to write the FEEL context to "destructure"/map the response

I'm missing a third option: A simple link to the documentation that shows how output mapping works, and what kind of assistance the user gets. I.e. they should get relevant completion as soon as they type a basic output context: `= { foo: | }` (and completion starts for available data.


---

Related to this we want to provide contextual editing based on examples:

![image](https://user-images.githubusercontent.com/58601/209323740-3a25277e-c9cb-4b1c-9426-d9d476d697f2.png)

You could argue that a basic example completion could be provided by the core editor, too (i.e. like a placeholder).

**Comment by christian-konrad** [comment]
> A simple link to the documentation that shows how output mapping works

We should definitely start with a simple doc link first in case the properties panel tooltips are not available when we consider this, this eases the pain and is easy to do. I'll create a separate issue

**Comment by MaxTru** [comment]
We need the supporting feature in prop panel: https://github.com/bpmn-io/properties-panel/issues/202. Afterwards we could provide connectors specific tooltips.

**Comment by sbuettner** [comment]
@nikku can we provide example data via the element templates? It would need to be dynamic as it might look different in case of connectors supporting multiple return types based on the selected operation.

**Comment by sbuettner** [comment]
As discussed in the architecture sessions:
* Start simple by fetching the first data example in the output dataclass of the ElementTemplate annotation and render it inside the tooltip.

**Comment by crobbins215** [comment]
🔗 Related context

**Scoped description:** Connector response/output mapping currently exposes `status`, `headers`, `body` (and, for REST, `document`) via a FEEL "Result Expression" (e.g. `= { foo: response.body.x }`), but the properties panel gives no in-app hint of these variable names or the FEEL destructuring syntax, and the resulting mapped variables don't appear in later autocompletion (confirmed against current docs). The team already agreed a direction in the existing thread: block on the properties-panel tooltip capability (bpmn-io/properties-panel#202), then implement via the existing `@DataExample` annotation (adding an optional `property` field) to render the first output-example from the ElementTemplate's output dataclass inside a tooltip, starting with the most prominent connectors (REST + Webhook). As of this check, that `property` field has not yet been added to `DataExample.java`, so the feature remains unimplemented. Note a separate, more recent architecture discussion is considering moving output mapping to the engine by default (away from being connector-internal), which could change how/whether this is ultimately built.
**Prior discussion:** This was discussed in #ask-connectors-team (Simon Büttner, Apr 2026) on output specification + autocompletion for output mapping via element templates, tracked in camunda/product-hub#2616; and in #prj-ai-native-builder-experience (Jan 2026) on moving output mapping to the engine as the default, referencing bpmn-io/bpmn-js-element-templates#157 ("Allow template authors to enable dynamic output mappings").

**💬 Suggested prompt for working this issue:**
> You are a Camunda Connectors engineer working on issue #154 in the connectors-experience backlog.
> **Issue:** Help tooltip for the response/output mapping in connectors
> **What's needed:** Connector response/output mapping currently exposes `status`, `headers`, `body` (and `document` for REST) via a FEEL Result Expression, but the properties panel gives no in-app hint of these variable names or FEEL syntax, and mapped variables aren't available in later autocompletion. Agreed direction (per issue history): once bpmn-io/properties-panel#202 lands, use the existing `@DataExample` annotation (add an optional `property` field) to surface the first output example from the ElementTemplate's output dataclass as a tooltip, starting with REST + Webhook connectors. Check whether the newer engine-level output-mapping direction (see Slack discussion) changes scope before implementing.
> **Relevant docs:** https://docs.camunda.io/docs/components/connectors/use-connectors/#variable-and-response-mapping, https://docs.camunda.io/docs/components/connectors/protocol/rest/#response
> **Relevant files:** element-template-generator/annotations/src/main/java/io/camunda/connector/generator/java/annotation/DataExample.java, element-template-generator/core/src/main/java/io/camunda/connector/generator/java/util/DataExampleModel.java, element-template-generator/core/src/main/java/io/camunda/connector/generator/java/ClassBasedDocsGenerator.java, connectors/http/http-base/src/main/java/io/camunda/connector/http/base/model/HttpCommonResult.java; UI work itself belongs in bpmn-io/properties-panel (blocked on #202) and possibly bpmn-io/bpmn-js-element-templates#157.
> **Constraints:** Follow existing code style. Flag anything uncertain rather than guessing.

---
_Generated by [Claude Code](https://claude.ai/code)_
